bitex/interface/bitfinex.py

```diff
@@ -55,9 +51,30 @@ def request(self, endpoint, authenticate=False, **req_kwargs):
 
     def _get_supported_pairs(self):
         """Return supported pairs."""
-        if self.REST.version == 'v1':
-            return self.symbols().json()
-        return requests.get('https://api.bitfinex.com/v1/symbols').json()
+        if self.REST.version == 'v1':  # default if v1
+            r = self.symbols()
+        else:
+            r = super(Bitfinex, self).request('GET', 'https://api.bitfinex.com/v1/symbols',
+                                              endpointwithversion=True)
+        if not r.ok:
+            raise PairFetchError(r.ok,r.status_code,r.reason)
+        pairs=r.json()
+        if not isinstance(pairs, list):
+            raise PairFetchError(pairs)
+
```

Comment:
this is an unnecessary check IMO. We code against the valid endpoint - if that breaks, we update our code accordingly. No need to support various variations, as there should be only one single point of truth.

Reply:
modified